src/main/java/graphql/execution/instrumentation/ChainedInstrumentation.java

```diff
@@ -349,18 +359,31 @@ public CompletableFuture<ExecutionResult> instrumentExecutionResult(ExecutionRes
     }
 
     static class ChainedInstrumentationState implements InstrumentationState {
-        private final Map<Instrumentation, InstrumentationState> instrumentationStates;
+        private final Map<Instrumentation, InstrumentationState> instrumentationToStates;
 
 
-        private ChainedInstrumentationState(List<Instrumentation> instrumentations, InstrumentationCreateStateParameters parameters) {
-            instrumentationStates = Maps.newLinkedHashMapWithExpectedSize(instrumentations.size());
-            instrumentations.forEach(i -> instrumentationStates.put(i, i.createState(parameters)));
+        private ChainedInstrumentationState(List<Instrumentation> instrumentations, List<InstrumentationState> instrumentationStates) {
+            instrumentationToStates = Maps.newLinkedHashMapWithExpectedSize(instrumentations.size());
+            for (int i = 0; i < instrumentations.size(); i++) {
+                Instrumentation instrumentation = instrumentations.get(i);
+                InstrumentationState instrumentationState = instrumentationStates.get(i);
+                instrumentationToStates.put(instrumentation, instrumentationState);
+            }
         }
 
         private InstrumentationState getState(Instrumentation instrumentation) {
-            return instrumentationStates.get(instrumentation);
+            return instrumentationToStates.get(instrumentation);
         }
 
+        private static CompletableFuture<InstrumentationState> combineAll(List<Instrumentation> instrumentations, InstrumentationCreateStateParameters parameters) {
+            Async.CombinedBuilder<InstrumentationState> builder = Async.ofExpectedSize(instrumentations.size());
+            for (Instrumentation instrumentation : instrumentations) {
+                // state can be null including the CF so handle that
+                CompletableFuture<InstrumentationState> stateCF = Async.orNullCompletedFuture(instrumentation.createStateAsync(parameters));
```

Comment:
we allow null to come back from instrumentation states inline with how it was able to be null before so we have `Async.orNullCompletedFuture` to handle this